xmodule/mongo_utils.py

```diff
@@ -59,11 +59,13 @@ def connect_to_mongodb(
         'port': port,
         'tz_aware': tz_aware,
         'document_class': dict,
+        # Only override the authSource parameter if it is extracted from the lowercased authsource kwarg above.
+        **({'authSource': auth_source} if auth_source else {})
         **kwargs,
     }
 
-    if user is not None and password is not None and not db.startswith('test_'):
-        connection_params.update({'username': user, 'password': password, 'authSource': db})
+    if user is not None and password is not None:
```

Comment:
The test database doesn't exist initially like a normal database so any call for authentication causes test failures.
That's why I've added an exemption of authentication for the test database.